src/index/src/inverted_index/search/index_apply/predicates_apply.rs

```diff
@@ -81,7 +86,8 @@ impl IndexApplier for PredicatesIndexApplier {
             bitmap &= bm;
         }
 
-        Ok(bitmap.iter_ones().collect())
+        output.matched_segment_ids = bitmap.iter_ones().collect();
```

Comment:
Maybe we could store matched_segment_ids in a `Vec` or use the bitmap as `matched_segment_ids`. The output of `iter_ones()` is sorted and we always iterate all segment ids. `Vec` should match what we require.

Reply:
Addressed